### ISSUE TITLE
Validator sometime reports problemResource, when removing @odata.type from StorageController

### ISSUE THREAD

**Issue by JojoWu19** [DMTF/Redfish-Service-Validator#294]
When we remove the `@odata.type` from all ReferenceableMember, the validator sometime reports problemResource on the StorageController. There are two StorageController elements and only one element is reported error, but they both should be correct.
P.S. Schema version is 2018.2

**Comment by tomasg2012** [comment]
If you could try to use the latest `master` branch to provide by a text and html file to read, I initially reviewed the debug log but could not capture the origin of the bug.

**Comment by JojoWu19** [comment]
@tomasg2012, the validator reports more ReferenceableMember errors (StorageController, PowerSupply and Fan) after I pull `master` and here is the log:
[logs.zip](https://github.com/DMTF/Redfish-Service-Validator/files/2580662/logs.zip)

**Comment by tomasg2012** [comment]
Seems this is a benign tool error, the resources reporting as errors are already checked earlier in the tool but as a different URI.  Will be fixed.

**Comment by tomasg2012** [comment]
Actually, upon second inspection, it appears your References are incorrect.

```
"Links": {
        "Chassis": [
            {
                "@odata.id": "/redfish/v1/Chassis/1"
            }
        ],
        "CooledBy": [
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/0"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/1"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/2"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/3"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/4"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/5"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/6"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/7"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/8"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Thermal/#/Fans/9"
            }
        ],
        "ManagedBy": [
            {
                "@odata.id": "/redfish/v1/Managers/1"
            }
        ],
        "PoweredBy": [
            {
                "@odata.id": "/redfish/v1/Chassis/1/Power/#/PowerSupplies/0"
            },
            {
                "@odata.id": "/redfish/v1/Chassis/1/Power/#/PowerSupplies/1"
            }
        ]
    },
```

This is an unusual error to catch so will have to be added to a corner case, but these URIs should not have a leading slash before the pound sign, at least, the tool considers it a completely different URI (might have to parse all URIs to determine their similarity).  The tool should report it, but this seems to be the tool's problem with interpreting what these resources are, given that the actual resources it works on don't have the leading slash.

**Comment by JojoWu19** [comment]
Yes, these URIs (PowerSupply and Fan) shall be corrected.
But it looks StorageController is a different case, right?

**Comment by tomasg2012** [comment]
Ah, my bad, appears that the logic that normally skips over these kinds of links doesn't catch this correctly.  By default, it would usually avoid "Links" until the target system is completely checked, but it doesn't catch this case.  This will be fixed as well.

**Comment by tomasg2012** [comment]
Please try the branch `links-fix`, if it solves the problem.

**Comment by JojoWu19** [comment]
This issue is fixed in the branch `links-fix`, thank you.

**Comment by JojoWu19** [comment]
Will this be merged back the branch master?
